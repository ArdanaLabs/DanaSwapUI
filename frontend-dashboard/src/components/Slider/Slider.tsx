import { Slider } from "@material-ui/core"
import { withStyles } from "@material-ui/core/styles"

const StyledSlider = withStyles((theme) => ({
  root: {
    maxWidth: "270px",
  },
  track: {
    height: 3,
    borderRadius: 2,
    color: "transparent",
  },
  thumb: {
    height: 16,
    width: 16,
    marginTop: 1.5,
    marginLeft: 0,
    color: theme.palette.secondary.main,
    transform: "translate(-50%, -50%)",
  },
  rail: {
    height: 3,
    background: `linear-gradient(-90deg, ${theme.palette.secondary.dark} 16.67%, ${theme.palette.secondary.main} 99.98%, ${theme.palette.secondary.dark} 99.99%)`,
    opacity: 1,
    borderRadius: 10,
  },
  mark: {
    background: "unset",
  },
  markLabel: {
    paddingTop: "5px",
    color: theme.palette.primary.main,
    fontFamily: "Museo Sans",
    fontStyle: "normal",
    fontWeight: 300,
    fontSize: "16px",
    lineHeight: "100%",
    [`&[data-index="0"]`]: {
      transform: "translateX(0%)",
    },
    [`&[data-index="1"]`]: {
      transform: "translateX(-100%)",
    },
  },
}))(Slider)

export default StyledSlider
