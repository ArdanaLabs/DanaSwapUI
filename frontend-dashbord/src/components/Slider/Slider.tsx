import { Slider } from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";

const StyledSlider = withStyles((theme) => ({
  root: {},
  track: {
    height: 6,
    borderRadius: 2,
    color: "transparent",
  },
  thumb: {
    height: 24,
    width: 24,
    marginTop: -9,
    marginLeft: -10,
    color: "#73D6F1",
  },
  rail: {
    height: "6px",
    background:
      "linear-gradient(-90deg, #2F3DA0 16.67%, #73D6F1 99.98%, #5F72FF 99.99%)",
    opacity: 1,
    borderRadius: 10,
  },
  mark: {
    background: "unset",
  },
  markLabel: {
    paddingTop: "5px",
    color: theme.palette.primary.main,
    fontFamily: "Museo Sans",
    fontStyle: "normal",
    fontWeight: 300,
    fontSize: "14px",
    lineHeight: "100%",
  },
}))(Slider);

export default StyledSlider;
