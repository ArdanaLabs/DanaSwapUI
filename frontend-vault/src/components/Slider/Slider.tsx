import { Slider } from "@material-ui/core"
import { withStyles } from "@material-ui/core/styles"

const StyledSlider = withStyles((theme) => ({
  root: {},
  track: {
    height: 6,
    borderRadius: 2,
    color: "transparent",
  },
  thumb: {
    height: 24,
    width: 24,
    marginTop: -11,
    marginLeft: -10,
    background: theme.palette.info.light,
  },
  rail: {
    height: "4px",
    background: theme.palette.info.dark,
    opacity: 1,
    borderRadius: 10,
  },
  mark: {
    background: "unset",
  },
  markLabel: {
    paddingTop: "5px",
    color: theme.palette.primary.main,
    fontFamily: "Museo Sans",
    fontStyle: "normal",
    fontWeight: 300,
    fontSize: "14px",
    lineHeight: "100%",
  },
}))(Slider)

export default StyledSlider
