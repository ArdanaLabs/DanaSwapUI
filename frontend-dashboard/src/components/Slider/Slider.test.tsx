import { Mark } from "@material-ui/core";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import * as Enzyme from "enzyme";
import { shallow } from "enzyme";
import Slider from ".";

Enzyme.configure({ adapter: new Adapter() });

let wrapper: Enzyme.ShallowWrapper;
const onAmountChange = jest.fn();
const marks: Mark[] = [
  {
    value: 0,
    label: "0%",
  },
  {
    value: 100,
    label: "100%",
  },
];

describe("Components / Slider / Slider", () => {
  beforeEach(() => {
    wrapper = shallow(
      <Slider
        min={0}
        max={100}
        defaultValue={0}
        value={12}
        onChange={onAmountChange}
        step={1}
        marks={marks}
      />
    );
  });

  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should be correct props", () => {
    expect(wrapper.prop("min")).toEqual(0);
    expect(wrapper.prop("max")).toEqual(100);
    expect(wrapper.prop("defaultValue")).toEqual(0);
    expect(wrapper.prop("value")).toEqual(12);
    expect(wrapper.prop("onChange")).toEqual(onAmountChange);
    expect(wrapper.prop("step")).toEqual(1);
    expect(wrapper.prop("marks")).toEqual(marks);
  });
});
