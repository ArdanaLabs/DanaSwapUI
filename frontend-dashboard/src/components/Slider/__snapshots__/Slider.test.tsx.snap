// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Slider / Slider renders 1`] = `ShallowWrapper {}`;
